src/crab_verifier.cpp

```diff
@@ -91,11 +92,15 @@ static checks_db generate_report(std::ostream& s,
         });
 
         if (thread_local_options.check_termination) {
-            bool pre_join_terminates = false;
-            for (const label_t& prev_label : bb.prev_blocks_set())
-                pre_join_terminates |= preconditions.at(prev_label).terminates();
+            m_db.max_instruction_count = INT_MAX;
+            for (const label_t& prev_label : bb.prev_blocks_set()) {
+                int instruction_count = preconditions.at(prev_label).get_max_instruction_count();
+                m_db.max_instruction_count = std::min(m_db.max_instruction_count, instruction_count);
+            }
```

Comment:
I think this is incorrect. A local variable is replaced here with a member, and the first assignment destroys the previous value of the member. `pre_join_terminates` is only for pinpointing the places where divergence might occur. It's not for deciding whether it occurs at all - this happens below. The global maximum instruction count is 
`max(db.max_instruction_count, from_inv.get_max_instruction_count())`. 
This is unrelated to the check on previous basic blocks, which still requires taking the minimum, using a local variable. 

Also, the name I suggested for the method might be confusing. Perhaps `get_instruction_count_upperbound` is a better name (since we take global maximum over upperbounds).

Reply:
updated, please check now.  I admit I had a hard time following the logic of the original code here.